### PR TITLE
[Windows] Fix OS in-place upgrade issues from Windows Server 2012 R2 and Windows Server 2016

### PR DESCRIPTION
**Pull request by Tomorrow9**
In Windows Server 2012 R2 or 2016 guest OS, get Windows Server 2025 image info as below:
```
TASK [Execute PowerShell command] **********************************************
task path: /home/worker/workspace/Ansible_Windows_Server_LTSC_Upgrade2016_MAIN_NVME_E1000E_EFI/ansible-vsphere-gos-validation/windows/utils/win_execute_cmd.yml:23
changed: [localhost -> test_vm] => {
    "changed": true,
    "cmd": "Get-WindowsImage -ImagePath D:\\sources\\install.wim",
    "delta": "0:00:21.125000",
    "end": "2024-05-10 09:50:20.468477",
    "rc": 0,
    "start": "2024-05-10 09:49:59.343477",
    "stderr": "",
    "stderr_lines": [],
....
    "stdout_lines": [
        "",
        "",
        "ImageIndex       : 1",
        "ImageName        : Windows Server 2025 SERVERSTANDARDCORE",
        "ImageDescription : Windows Server 2025 SERVERSTANDARDCORE",
        "ImageSize        : 8,953,831,686 bytes",
        "",
        "ImageIndex       : 2",
        "ImageName        : Windows Server 2025 SERVERSTANDARD",
        "ImageDescription : Windows Server 2025 SERVERSTANDARD",
        "ImageSize        : 18,945,747,354 bytes",
        "",
        "ImageIndex       : 3",
        "ImageName        : Windows Server 2025 SERVERDATACENTERCORE",
        "ImageDescription : Windows Server 2025 SERVERDATACENTERCORE",
        "ImageSize        : 8,949,864,520 bytes",
        "",
        "ImageIndex       : 4",
        "ImageName        : Windows Server 2025 SERVERDATACENTER",
        "ImageDescription : Windows Server 2025 SERVERDATACENTER",
        "ImageSize        : 18,946,223,263 bytes",
        "",
        "",
        ""
    ]
```

### CODE REVIEW

**Comment by 123lzxm** [comment]
How about Windows Server 2012 R2? What's the image info got from it?

File: windows/guest_os_inplace_upgrade/check_after_gos_upgrade.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ---
 - name: "Set fact of expected guest OS distribution"
   ansible.builtin.set_fact:
-    win_distr_expected: "{{ 'Microsoft ' ~ win_image_name.split('(')[0].strip() }}"
+    win_distr_expected: "{{ 'Microsoft ' ~ win_image_name | regex_search('(Windows|Windows Server) ([0-9]+)') ~ ' ' ~ guest_os_edition }}"
```
[the remark this answers is not in the document]
Comment (Tomorrow9) [reply windows/guest_os_inplace_upgrade/check_after_gos_upgrade.yml]: Right, thanks.

**Comment by Tomorrow9** [comment]
> How about Windows Server 2012 R2? What's the image info got from it?

It's the same as Windows Server 2016.

File: windows/utils/win_get_driver_installer.yml [hunk 2]
```diff
@@ -24,19 +24,26 @@
...
       if ($(Test-Path -Path "HKLM:\System\CurrentControlSet\Services\{{ win_driver_service }}")) {
-          Get-ItemPropertyValue -Path "HKLM:\System\CurrentControlSet\Services\{{ win_driver_service }}" -Name 'vwdk.installers'
+        reg query "HKLM\System\CurrentControlSet\Services\{{ win_driver_service }}" /v 'vwdk.installers' | findstr 'REG'
       }
+
+#   win_powershell_cmd: |-
+#     if ($(Test-Path -Path "HKLM:\System\CurrentControlSet\Services\{{ win_driver_service }}")) {
+#         Get-ItemPropertyValue -Path "HKLM:\System\CurrentControlSet\Services\{{ win_driver_service }}" -Name 'vwdk.installers'
+#     }
+
```
Comment (keirazhang) [inline windows/utils/win_get_driver_installer.yml:40]: Shall these lines be removed?

File: windows/utils/get_windows_system_info.yml [hunk 3]
```diff
@@ -33,10 +33,11 @@
   block:
     - name: "Set fact of Windows guest OS edition info"
       ansible.builtin.set_fact:
-        guest_os_edition_info: "{{ guest_os_ansible_distribution | regex_search('Microsoft (Windows|Windows Server) ([0-9]+) (.*)', '\\3') }}"
+        guest_os_edition_info: >-
+          {{ guest_os_ansible_distribution | regex_search('Microsoft Windows( Server)? ([0-9]+) (R2 )?(.*)', '\4') }}
```
Comment (keirazhang) [inline windows/utils/get_windows_system_info.yml:37]: regex_search or regex_replace?

File: windows/utils/win_get_driver_installer.yml (hunk 2 above)
Comment (Tomorrow9) [reply windows/utils/win_get_driver_installer.yml:40]: Yeah, removed, thanks.

File: windows/utils/get_windows_system_info.yml (hunk 3 above)
Comment (Tomorrow9) [reply windows/utils/get_windows_system_info.yml:37]: Add `first` filter to get the one in the match list. Thanks.

**Review by keirazhang — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**